common/src/chain/transaction/signature/tests/utils.rs

```diff
@@ -46,26 +48,30 @@ impl MutableTransaction {
 }
 
 pub fn generate_unsigned_tx(
-    outpoint_dest: Destination,
-    inputs_count: u32,
-    outputs_count: u32,
+    destination: &Destination,
+    inputs_count: usize,
+    outputs_count: usize,
 ) -> Result<Transaction, TransactionCreationError> {
     let mut rng = rand::thread_rng();
-    let mut inputs = Vec::new();
-    for input_index in 0..inputs_count {
-        inputs.push(TxInput::new(
-            Id::<Transaction>::new(&H256::random()).into(),
-            input_index,
-            InputWitness::NoSignature(None),
-        ));
-    }
-    let mut outputs = Vec::new();
-    for _ in 0..outputs_count {
-        outputs.push(TxOutput::new(
+
+    let inputs = (0..inputs_count)
+        .map(|input_index| {
+            TxInput::new(
+                Id::<Transaction>::new(&H256::random()).into(),
+                input_index as u32,
+                InputWitness::NoSignature(None),
+            )
+        })
+        .collect();
+
+    let outputs = std::iter::from_fn(|| {
+        Some(TxOutput::new(
             Amount::from_atoms(rng.gen::<IntType>()),
-            outpoint_dest.clone(),
-        ));
-    }
+            destination.clone(),
+        ))
+    })
+    .take(outputs_count)
+    .collect();
 
     let tx = Transaction::new(0, inputs, outputs, 0)?;
```

Comment:
The flags and lock_time being fixed to 0 simply tell me that you haven't tested mutating these.